resources/aws_ec2.go

```diff
@@ -820,9 +849,60 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 		}
 		log.Printf("%s: instance running", obj)
 	}
+	// save the public ip to the struct for resources to receive.
+	obj.PublicIPv4, err = getPublicIPv4(obj.client, instanceID)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error getting public ip")
+	}
+	// save the private ip to the struct for resources to receive.
+	obj.PrivateIPv4, err = getPrivateIPv4(obj.client, instanceID)
+	if err != nil {
+		return false, errwrap.Wrapf(err, "error getting public ip")
+	}
+
 	return false, nil
 }
 
+// getPublicIPv4 gets the instance's public ip address. If the instance has a
+// public ip address it is returned (as a string), and if it does not, the
+// method returns an empty string.
+func getPublicIPv4(c *ec2.EC2, instanceID *string) (string, error) {
```

Comment:
Since we're basically making the same call for this and `getPrivateIPv4`, why not just make one generic getInstance call, and then process the output? (Or if you really want to be specific, return the instanceInfo struct (or whatever it's called)